contracts/NodeOperatorRegistry.sol

```diff
@@ -574,75 +587,115 @@ contract NodeOperatorRegistry is
         uint256 _index,
         bytes memory _proof
     ) external override whenNotPaused {
-        uint256 operatorId = operatorOwners[msg.sender];
-        require(operatorId != 0, "Operator doesn't exist");
-
+        uint256 operatorId = getOperatorId(msg.sender);
         NodeOperator storage no = operators[operatorId];
 
-        require(
-            no.status == NodeOperatorStatus.CLAIMED,
-            "Operator status isn't CLAIMED"
+        checkStatus(
+            no.status,
+            NodeOperatorStatus.CLAIMED,
+            "The Operator status isn't CLAIMED"
         );
 
-        require(_proof.length != 0, "Empty proof");
-        require(_accumFeeAmount != 0, "AccumFeeAmount is ZERO");
-        require(_index != 0, "index is ZERO");
-
-        IValidator(no.validatorContract).claimFee(
+        address rewardAddress = no.rewardAddress;
+        IValidator(no.validatorProxy).claimFee(
             _accumFeeAmount,
             _index,
-            _proof
+            _proof,
+            rewardAddress
         );
 
         totalClaimedNodeOperator--;
         totalExitNodeOperator++;
         no.status = NodeOperatorStatus.EXIT;
-        no.statusTimestamp = block.timestamp;
+        no.statusUpdatedTimestamp = block.timestamp;
+
+        emit ClaimFee(operatorId, rewardAddress);
+    }
+
+    /// @notice Allows the operator's owner to withdraw rewards.
+    function withdrawRewards() external override whenNotPaused {
+        uint256 operatorId = getOperatorId(msg.sender);
+        NodeOperator memory no = operators[operatorId];
+        checkStatus(
+            no.status,
+            NodeOperatorStatus.ACTIVE,
+            "The Operator status isn't ACTIVE"
+        );
 
-        emit ClaimFee(
-            operatorId,
+        address rewardAddress = no.rewardAddress;
+        uint256 rewards = IValidator(no.validatorProxy).withdrawRewards(
             no.validatorId,
-            _accumFeeAmount,
-            _index,
-            _proof
+            rewardAddress
         );
+
+        emit WithdrawRewards(operatorId, rewardAddress, rewards);
     }
 
-    /// @notice Allows to update signer publickey
+    /// @notice Allows the operator's owner to update signer publickey.
     /// @param _signerPubkey new signer publickey
     function updateSigner(bytes memory _signerPubkey)
         external
         override
         whenNotPaused
     {
-        uint256 operatorId = operatorOwners[msg.sender];
-        require(operatorId != 0, "Operator doesn't exist");
-
+        uint256 operatorId = getOperatorId(msg.sender);
         NodeOperator storage no = operators[operatorId];
-
         require(
-            no.status == NodeOperatorStatus.STAKED,
-            "Operator status not STAKED"
+            no.status <= NodeOperatorStatus.ACTIVE,
+            "The Operator status isn't ACTIVE"
```

Comment:
Rewrite this message. It is not accurate, since this modifier allows INACTIVE status.

Reply:
No, it's correct because we allows the operator'owner to set the signer key when they did not staked yet.